suite/suite.go

```diff
@@ -32,8 +58,44 @@ func (suite *Suite) SetT(t *testing.T) {
 }
 
 // Run takes a testing suite and runs all of the tests attached
-// to it.
+// to it.  For legacy reasons, Run uses a default test name format
+// of "{method}", which only prints out the test method name.  See
+// NamedRun for a version of Run that accepts other format strings
+// for the test name in testing output.
 func Run(t *testing.T, suite TestingSuite) {
+	run("", t, suite)
+}
+
+// NamedRun performs a test run the same as Run, but with a different
+// test name format.  Pass a pattern to use as your test names for
+// nameFormat.  The following patterns will be replaced by details
+// about the current test:
+//
+// * "{function}": The name of the function that called NamedRun.
+// * "{suite}": The name of the testing suite's underlying type.
+// * "{method}": The name of the test method.
+//
+// Be aware that the test flag "nameformat" will override nameFormat
+// globally.
+func NamedRun(nameFormat string, t *testing.T, suite TestingSuite) {
```

Comment:
Assertions' signatures start with `*testing.T`, so I would move `t` to the first argument for consistency.

Talking about consistency, I would consider moving the name to be the final argument since it's more consistent with assertions allowing customising the output message.

As a final point, without reading the docs I would expect NamedRun to prefix the method names rather than providing a new format. I would suggest `NamedRun` to call `run(name + ".{method}", t, suite)` and have a different method to pass on a format.